src/dotnet/commands/dotnet-compile-native/CompileNativeCommand.cs

```diff
@@ -0,0 +1,149 @@
+            Directory.CreateDirectory(nativeOutputPath);
+            Directory.CreateDirectory(nativeIntermediateOutputPath);
+
+            var compilationOptions = context.ProjectFile.GetCompilerOptions(context.TargetFramework, args.ConfigValue);
+            var managedOutput = outputPathCalculator.GetAssemblyPath(args.ConfigValue);
+
+            var nativeArgs = new List<string>();
+
+            // Input Assembly
+            nativeArgs.Add($"{managedOutput}");
+            // ILC Args
+            if (!string.IsNullOrWhiteSpace(args.IlcArgsValue))
+            {
+                nativeArgs.Add("--ilcargs");
+                nativeArgs.Add($"{args.IlcArgsValue}");
+            }
+
+            // ILC Path
+            if (!string.IsNullOrWhiteSpace(args.IlcPathValue))
+            {
+                nativeArgs.Add("--ilcpath");
+                nativeArgs.Add(args.IlcPathValue);
+            }
+
+            // ILC SDK Path
+            if (!string.IsNullOrWhiteSpace(args.IlcSdkPathValue))
+            {
+                nativeArgs.Add("--ilcsdkpath");
+                nativeArgs.Add(args.IlcSdkPathValue);
+            }
+
+            // AppDep SDK Path
+            if (!string.IsNullOrWhiteSpace(args.AppDepSdkPathValue))
+            {
+                nativeArgs.Add("--appdepsdk");
+                nativeArgs.Add(args.AppDepSdkPathValue);
+            }
+
+            // CodeGen Mode
+            if (args.IsCppModeValue)
+            {
+                nativeArgs.Add("--mode");
+                nativeArgs.Add("cpp");
+            }
+
+            if (!string.IsNullOrWhiteSpace(args.CppCompilerFlagsValue))
+            {
+                nativeArgs.Add("--cppcompilerflags");
+                nativeArgs.Add(args.CppCompilerFlagsValue);
+            }
+
+            // Configuration
+            if (args.ConfigValue != null)
+            {
+                nativeArgs.Add("--configuration");
+                nativeArgs.Add(args.ConfigValue);
+            }
+
+            // Architecture
+            if (args.ArchValue != null)
+            {
+                nativeArgs.Add("--arch");
+                nativeArgs.Add(args.ArchValue);
+            }
+
+            // Intermediate Path
+            nativeArgs.Add("--temp-output");
+            nativeArgs.Add($"{nativeIntermediateOutputPath}");
+
+            // Output Path
+            nativeArgs.Add("--output");
+            nativeArgs.Add($"{nativeOutputPath}");
+
+            // Write Response File
+            var rsp = Path.Combine(nativeIntermediateOutputPath, $"dotnet-compile-native.{context.ProjectFile.Name}.rsp");
+            File.WriteAllLines(rsp, nativeArgs);
+
+            // TODO Add -r assembly.dll for all Nuget References
+            //     Need CoreRT Framework published to nuget
+
+            // Do Native Compilation
+            var exitCode = CompileNativeCommandInternal.Run(new string[] { "--rsp", $"{rsp}" });
```

Comment:
I think you could just do `.Run(nativeArgs)` rather than writing the response file here. Now that this isn't a separate process not sure this is necessary.